lib/ansible/modules/web_infrastructure/ansible_tower/tower_host.py

```diff
@@ -108,18 +119,30 @@ def main():
     tower_auth = tower_auth_config(module)
     with settings.runtime_values(**tower_auth):
         tower_check_mode(module)
-        host = tower_cli.get_resource('host')
 
         try:
             inv_res = tower_cli.get_resource('inventory')
+            group_res = tower_cli.get_resource('group')
+            host_res = tower_cli.get_resource('host')
             inv = inv_res.get(name=inventory)
 
             if state == 'present':
-                result = host.modify(name=name, inventory=inv['id'], enabled=enabled,
-                                     variables=variables, description=description, create_on_missing=True)
+                result = host_res.modify(name=name, inventory=inv['id'], enabled=enabled,
+                                         variables=variables, description=description, create_on_missing=True)
+                host = host_res.get(name=name, inventory=inv['id'])
+                if groups:
+                    changed = {'changed': False}
+                    for group in groups:
+                        group = group_res.get(inventory=inv['id'], name=group)
+                        changed = host_res.associate(host=host['id'], group=group['id'])
+                    json_output['groups_added'] = changed['changed']
```

Comment:
Let's say I have 2 groups in `groups`. The 2nd entry is already associated with the host, but the 1st entry is not yet associated with the host.

In that case, the `changed` dictionary shows `{"changed": True}`, but then flips to False, and the task eventually returns a status of changed=False, incorrectly.

Instead, inside the loop you should check if changed=True, and then set `json_output['groups_added']` to True. Also, if an association happened, shouldn't the task return a changed status as well?

Reply:
@AlanCoding I believe i made a mistake and misunderstood one of the error messages.. i believe i've fixed that now but the PR appears to have been merged with 0 commits...